pkg/controller/common/reconciler/secret.go

```diff
@@ -10,7 +10,20 @@ import (
 	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
 	"github.com/elastic/cloud-on-k8s/pkg/utils/maps"
 	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Labels set on secrets which cannot rely on owner references due to https://github.com/kubernetes/kubernetes/issues/65200,
+// but should still be garbage-collected (best-effort) by the operator upon owner deletion.
+const (
+	SoftOwnerNamespaceLabel = "eck.k8s.elastic.co/soft-owner-namespace"
+	SoftOwnerNameLabel      = "eck.k8s.elastic.co/soft-owner-name"
+	SoftOwnerKindLabel      = "eck.k8s.elastic.co/soft-owner-kind"
```

Comment:
I think we could lose the `soft` bit and it would still be clear that it is an alternative to the built-in owner reference mechanism. Don't feel strongly though.

Reply:
Do you mean in the label name, or throughout the code?
I think I agree for the label part; however in the code having `Soft` makes things more explicit I think (this is not the namespace of the owner in k8s sense, it's the namespace of our soft owner in eck sense).
I think I'll remove `soft` from the label name but keep it in the code structures. WDYT?